### PR TITLE
Create file structure with crates and connection submodule (Issue #1)

### LINKED ISSUE #1: Create file structure with crates and connection submodule
Set up the initial file structure for the datajoint-core repository. This repository should contain two Rust library crates (packages) and should follow standard conventions for organizing both. The following two packages should be contained in the repository:

- **datajoint-core** - A Rust library for shared code across DataJoint clients. Will currently facilitate connections to SQL databases and raw querying.
- **datajoint-core-ffi-c** - A Rust library for calling datajoint-core from other languages using the C FFI. This crate defines all C-interface functions (written in Rust) that expose the functionality of datajoint-core to other languages.

The datajoint-core package should currently contain only one submodule: the `connection` submodule. This submodule should somehow be accessible to the outside world (for exmaple: `datajoint_core::connection`). In the future, datajoint-core will support more submodules as more functionality is ported over, such as `query`, `schema`, and more.

Both Rust packages should be usable as a library crate that can be used by other Rust programs.

Here is an example structure that puts these two packages in one repository:
```
/packages
    /datajoint-core
        /src
            /connection
        Cargo.toml
    /datajoint-core-ffi-c
        /src
        Cargo.toml
LICENSE
README.md
```

### PR DESCRIPTION
**Pull request by JHocevar**
Closes #1

### CODE REVIEW

**Comment by Rhijke** [comment]
@JHocevar can you change the name of the PR to 'Create file structure with crates and connection submodule (Issue #1)' so it's easier to track later?

**Review by jackson-nestelroad — DISMISSED** [review]
LGTM. A lot of this code will likely be replaced by actual library code, but file structure and everything interconnecting is great.

**Review by guzman-raphael — APPROVED** [review]
@JHocevar Nice job on your first PR! :handshake:
Only have some minor suggestions regarding styling and docs.

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
Whoops, sorry. Submitted the review too early.

File: datajoint-python/connection.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from cffi import FFI
+ffi = FFI()
+
+ffi.cdef("""
+    void conn();
+    """)
```
[the remark this answers is not in the document]
Comment (JHocevar) [reply datajoint-python/connection.py:6]: Thanks for the link to the guidelines. I went back and forth a few ways on that one, and couldn't decide which format looked best.

**Review by guzman-raphael — APPROVED** [review]
@JHocevar Thanks for the updates! :rocket:

**Pull request merged**